### PR TITLE
DOP-4441: Improve snooty.clickInclude command in VSCode extension

### PR DESCRIPTION
**Pull request by mayaraman19**
[DOP-4441](https://jira.mongodb.org/browse/DOP-4441)

Enables the `clickInclude` function to work properly. Also includes updates to documentation to get the extension running locally, since I had a hard time getting it set up. Works with branch [DOP-4441](https://github.com/mongodb/snooty-parser/pull/594) of the parser.

To test, checkout this branch locally as well as the corresponding parser branch. Set up your local extension as mentioned in the instructions for this project's `README` on some content repo (I used `docs-compass`). Hover + click on different filenames mentioned  - `.txt` files, `.rst` files that don't exist in the `/includes/` folder, `.png` and other image files, files that don't have an extension.

**Notes:**

I don't know why it marked so much of the file as having been changed :sob:, but I have left comments on which parts of the file I actually changed.

### CODE REVIEW

File: src/extension.ts
Comment (mayaraman19) [inline src/extension.ts:119]: change here

File: src/extension.ts
Comment (mayaraman19) [inline src/extension.ts:159]: added these lines

File: src/extension.ts
Comment (mayaraman19) [inline src/extension.ts:179]: added/changed these lines

File: src/extension.ts
Comment (rayangler) [inline src/extension.ts:148]: (Non-blocking comment) Based on the generic regex used for hovering, doc role links are captured and the resulting hover content ends up looking a little weird (see image). This is most likely out of scope since I think this is partially existing behavior (with the long query param being a new addition). 

Just wanted to call this out in case this is something that seems like we should file a separate ticket for to fix.

Screen Shot 2024-05-13 at 6 38 50 PM

File: README.md
Comment (mayaraman19) [reply README.md:29]: I'm not sure - I'm not a big function key user, but when testing it seems like this method doesn't work:
image

File: src/extension.ts
Comment (mayaraman19) [reply src/extension.ts:158]: For context, I added this to support files under the `.. toctree::` directive (example [here](https://github.com/mongodb/docs-compass/blob/master/source/collections.txt#L152)) because these files are usually written without the extension and are always `.txt` files, and without adding this extension the parser would error out. Open to any suggestions, though, if you think this behavior shouldn't happen

File: README.md
Comment (i80and) [reply README.md:29]: We should absolutely fix the issue preventing the debug button from working. It's a small one line fix https://jira.mongodb.org/browse/DOP-4705

File: src/extension.ts
Comment (mayaraman19) [reply src/extension.ts:148]: thanks for catching this! will file a ticket once this one is merged

File: README.md
Comment (rayangler) [reply README.md:29]: Gotcha, thanks! Idk why it was working on my end when I tried last week. 🤔 

I think it might be confusing to have 2 different sections on how to run the extension locally in 2 different places in the repo. It might be good to consolidate the instructions with clarity on the differences, but that could always be included in the above ticket

**Review by rayangler — APPROVED** [review]
LGTM after merging parser PR

**Pull request merged**
